### PR TITLE
Add ability for separate stroke layer for vector legends

### PR DESCRIPTION
**Pull request by bartvde**
SDK-826

### CODE REVIEW

File: src/components/legend.js [hunk 1]
```diff
@@ -225,7 +225,21 @@ export class Legend extends React.Component {
...
+    let strokeEqual = true;
+    if (this.props.strokeId) {
+      const nextStrokeLayer = getLayerById(nextProps.layers, this.props.strokeId);
+      const strokeLayer = getLayerById(this.props.layers, this.props.strokeId);
+      strokeEqual = jsonEquals(strokeLayer, nextStrokeLayer);
+    }
+    return !layerEqual || !strokeEqual;
+  }
+
+  getVectorLegend(layer, layer_src, strokeLayer) {
```
Comment (theduckylittle) [inline src/components/legend.js]: Should this pass in an array of layers instead of enumerating two of them?

Comment (bartvde) [reply src/components/legend.js]: We can, if we always assume the second one is the stroke layer

Comment (bartvde) [reply src/components/legend.js]: Actually, while implementing, the code would become more complex actually, so I'm not sure if it's a good idea

Comment (bartvde) [reply src/components/legend.js]: I tried to implement that approach, but the code actually became more complex and longer by doing so

**Comment by bartvde** [comment]
@theduckylittle I did restructure it a bit according to your comment see https://github.com/boundlessgeo/sdk/pull/792/commits/c41a0f4c3da64dda69efad9f91b037faa5c26a86

**Pull request merged**